addons/hr/models/hr_departure_reason.py

```diff
@@ -12,17 +12,35 @@ class HrDepartureReason(models.Model):
 
     sequence = fields.Integer("Sequence", default=10)
     name = fields.Char(string="Reason", required=True, translate=True)
-    reason_code = fields.Integer()
+    country_code = fields.Char("Country", default=lambda self: self.env.company.country_code)
+    in_current_countries = fields.Boolean("Country matches a selected company", search='_search_in_current_countries', store=False)
 
+    @api.model
+    def _countries_domain(self):
+        country_codes = self.env.companies.mapped('country_code')
+        return [('country_code', 'in', [*country_codes, False])]
```

Comment:
```suggestion
        return [('country_code', 'in', country_codes + [False])]
```
usually we write that :)